internal/controller/controllers/agent_controller.go

```diff
@@ -516,6 +506,50 @@ func (r *AgentReconciler) shouldApproveCSRsForAgent(ctx context.Context, agent *
 	return false, nil
 }
 
+func marshalNodeLabels(nodeLabels map[string]string) (string, error) {
+	b, err := json.Marshal(&nodeLabels)
+	if err != nil {
+		return "", err
+	}
+	return string(b), err
+}
+
+func (r *AgentReconciler) applyDay2NodeLabels(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent, client spoke_k8s_client.SpokeK8sClient) error {
+	if !funk.IsEmpty(agent.Spec.NodeLabels) {
```

Comment:
IMO it's cleaner to do 
```
if funk.IsEmpty(agent.Spec.NodeLabels) {
	return nil
}
```
and have this block of code outside the if

Reply:
OK